server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -102,6 +110,7 @@ protected void afterExecute(Runnable r, Throwable t) {
             executionEWMA.addValue(taskExecutionNanos);
             totalExecutionTime.add(taskExecutionNanos);
         }
+        ongoingTasks.remove(r);
```

Comment:
I think it's better to wrap the whole body of the function in a `try...finally` and put this line in the finally part. Because I see that `super.afterExecute(r, t);` may throw, and possibly who knows other stuff may throw, so it is better to ensure that the removal happens with the finally part.

Reply:
It does sound safer. Pushed [bb63339](https://github.com/elastic/elasticsearch/pull/97661/commits/bb63339ff9b0d160cfc4d14d312bc9d9bf31fdff).